osu.Game/OsuGame.cs

```diff
@@ -758,6 +758,10 @@ private void showOverlayAboveOthers(OverlayContainer overlay, OverlayContainer[]
         {
             otherOverlays.Where(o => o != overlay).ForEach(o => o.Hide());
 
+            // generally shouldn't ever hit this state, but protects against a crash on attempting to change ChildDepth.
+            if (overlay.LoadState < LoadState.Ready)
```

Comment:
Should this be `<=`? I assume this is coming from a `LoadComponentAsync(overlay, overlayContainer.Add)`, in which case there's a chance that this is invoked in the 1 particular frame where the it's in the `Ready` state.

Or, better still, you could rewrite this code as:
```
// Partially visible so leave it at the current depth.
if (overlay.IsPresent)
    return;

// Show above all other overlays.
if (overlay.IsLoaded) // Or overlay.Parent != null...
    overlayContainer.ChangeChildDepth(overlay, (float)-Clock.CurrentTime);
else
    overlay.Depth = (float)-Clock.CurrentTime;
```
Which also takes care of another edgecase of the overlay coming visible after it finishes loading. I'm not sure how overlays behave in this scenario in general, but it could mean that the overlay appears below the others.

Reply:
Yeah, I guess it needs to be `> Ready` to be able to make the `ChangeChildDepth` call doesn't it.

Feel free to make that change. This shouldn't be hit either way; it's more of a safety thing.